lisa/tools/ethtool.py

```diff
@@ -321,51 +321,69 @@ def get_device_link_settings(self, interface: str) -> DeviceLinkSettings:
         return device_link_settings
 
     def get_device_ring_buffer_settings(
-        self, interface: str
+        self, interface: str, force: bool = False
     ) -> DeviceRingBufferSettings:
-        if interface in self._device_ring_buffer_settings_map.keys():
+        if (not force) and (interface in self._device_ring_buffer_settings_map.keys()):
             return self._device_ring_buffer_settings_map[interface]
 
         result = self.run(f"-g {interface}")
         if result.exit_code != 0:
```

Comment:
there is a new method in TestResult, called `assert_exit_code`, it can be used to simplify the code.